src/static/clientConfig.ts

```diff
@@ -53,6 +54,8 @@ export default class ClientConfig<Params extends BaseUriParameters>
     ClientConfigInit<Params>['transformRequest']
   >;
 
+  public throwOnBadResponse?: boolean = false;
```

Comment:
```suggestion
  public throwOnBadResponse: boolean;
```
Makes life easier if you have fewer possible states to deal with. 😉 

(Should remain optional in `ClientConfigInit`.)